### DISCUSSION TITLE
Will harp.gl keep opensource after HERE stopped the engagement

### DISCUSSION THREAD

**Discussion by luolanaaTUD** [heremaps/harp.gl#2385] [General]
"HERE is stopping its engagement on Harp.gl starting 03/01/2022 in favour of a fully productised and production-grade integration of the harp.gl rendering engine into HERE Maps API for JavaScript (https://developer.here.com/develop/javascript-api) All 3D features and many more will be offered via HERE Maps API for JavaScript moving forward." 

Will harp.gl keep updating and open to the public?

**Comment by catnuko** [comment]
harp. gl has great potential in front end 3D maps

**Comment by nzjony** [comment]
It will stay open, i.e. it can still be forked but it won't be updated anymore.

**Reply by trusktr** [reply]
I think a better question is, will the replacement (HERE JS API) be open source? Or will it be closed source like Google Maps?

**Reply by nzjony** [reply]
@trusktr , HERE JS API is closed source

**Comment by trusktr** [comment]
The website no longer loads. 😭

**Reply by trusktr** [reply]
I will continue efforts at https://github.com/lume/harp.gl. I will make a new site, and will also be adding a LUME HTML interface for Harp.gl that will look something like the following:

```html
<lume-harp-scene>
  <!-- position a node with geospatial coordinates -->
  <lume-harp latlonalt="100 100 100">
    <!-- place any 3D LUME elements relative to the Harp node using LUME APIs. -->
    <lume-box size="100 100 100" color="pink" rotation="10 20 30" ...></lume-box>
  <lume-harp>
<lume-harp-scene>
```

Layering will be as simple as this:

```html
<lume-harp-scene>
  <!-- this layer is visible -->
  <lume-node visible="true">
    <lume-harp latlonalt="100 100 100">
      <lume-box size="100 100 100" color="pink" rotation="10 20 30" ...></lume-box>
    <lume-harp>
    <lume-harp latlonalt="...">
      <lume-sphere size="100 100 100" color="pink" rotation="10 20 30" ...></lume-sphere>
    <lume-harp>
  </lume-node>

  <!-- this layer is not visible -->
  <lume-node visible="false">
    <lume-harp latlonalt="100 100 100">
      <lume-gltf-model src="/path/to/model.gltf" rotation="10 20 30" ...></lume-gltf-model>
    <lume-harp>
    <lume-harp latlonalt="...">
      <lume-gltf-model src="/path/to/other.gltf" rotation="10 20 30" ...></lume-gltf-model>
    <lume-harp>
  </lume-node>
<lume-harp-scene>
```

Any regular 2D DOM ui can be added on top to control visibility of a set of LUME nodes being used for layering.

This will be easy to use in React, Vue, Svelte, Solid.js, Angular, jQuery, pick-your-favorite-dom-manipulation-library, and the HTML interface will not require any build setup (for example easy pasting of HTML onto Codepen, or plain HTML files).

**Reply by nzjony** [reply]
Thanks for the update @trusktr !

**Reply by trusktr** [reply]
Some initial plans for the fork:

- get a website back up
- get the lume interface working
- add some features for simplicity like declarative map markers

Longer term:

- port Harp to LUME elements and Solid reactivity for declarative simplicity
  - makes the system compatible with server side rendering, and the declarative HTML search-engine friendly such that URLs to locations on the map can be indexed if the output contains text (even 3D WebGL text)
- finish [lume/glas](https://github.com/lume/glas) features needed for Harp (Three.js port to AssemblyScript that compiles to WebAssembly)
- port the LUME/Harp system on top of that to AS and compile the whole thing to WebAssembly
